project-chip/connectedhomeip
# [Telink]: BLE commissioning (source part)

#### Problem
Currently Zephyr OS does not contain implemented Telink BLE controller yet.
So that, BLE functionality for Matter project is implemented with Telink BLE SDK.
This is Source code part. Docker PR here: https://github.com/project-chip/connectedhomeip/pull/15722
Docker PR shall be merged, built and pushed before This PR merge.
As soon new image will be built, new docker version will be updated in .github/workflows as well 

#### Change overview
* Add Telink BLE SDK in third_party/telink_sdk
* Implement BLE functionality for Matter in BLEManagerImpl.cpp
* Add button and handler to Open Basic commissioning window in AppTask.cpp of lighting-app example
* Other fixes and cleanups

#### Testing
Tested manually with chip-tool.
Steps: 
* Run: $ chip-tool pairing ble-thread <...>
* Wait till success
* Run: $ chip-tool onoff <...>
* Check if end device perform command
